src/main/scala/io/qbeast/table/IndexedTable.scala

```diff
@@ -284,23 +283,25 @@ private[table] class IndexedTableImpl(
     val isNewCubeSize = latestRevision.desiredCubeSize != qbeastOptions.cubeSize
     // Checks if the user-provided column boundaries would trigger the creation of
     // a new revision.
-    val isNewSpace = qbeastOptions.stats match {
+    val isNewSpace = qbeastOptions.columnStats match {
       case None => false
-      case Some(stats) =>
-        val columnStats = stats.first()
-        val transformations = latestRevision.transformations
-
+      case Some(statsString) =>
+        val spark = SparkSession.active
+        import spark.implicits._
+        val columnStatsRow = spark.read
```

Comment:
Can we extract this code block in a method somewhere? I've noticedon `SparkRevisionFactory` too

Reply:
This will be taken care of together with #223